core/trino-main/src/test/java/io/trino/execution/TestSetTimeZoneTask.java

```diff
@@ -0,0 +1,172 @@
+        accessControl = new AllowAllAccessControl();
+
+        metadata = createTestMetadataManager(transactionManager, new FeaturesConfig());
+    }
+
+    @AfterClass(alwaysRun = true)
+    public void tearDown()
+    {
+        executor.shutdownNow();
+        executor = null;
+    }
+
+    @Test
+    public void testSetTimeZoneLocal()
+    {
+        QueryStateMachine stateMachine = createQueryStateMachine("SET TIME ZONE LOCAL");
+        SetTimeZone setTimeZone = new SetTimeZone(Optional.empty());
+        executeSetTimeZone(setTimeZone, stateMachine);
+
+        Map<String, String> setSessionProperties = stateMachine.getSetSessionProperties();
+        assertThat(setSessionProperties).hasSize(1);
+        assertEquals(setSessionProperties.get(TIME_ZONE_ID), "America/Bahia_Banderas");
+    }
+
+    @Test
+    public void testSetTimeZoneStringLiteral()
+    {
+        QueryStateMachine stateMachine = createQueryStateMachine("SET TIME ZONE 'America/Los_Angeles'");
+        SetTimeZone setTimeZone = new SetTimeZone(Optional.of(new StringLiteral("America/Los_Angeles")));
+        executeSetTimeZone(setTimeZone, stateMachine);
+
+        Map<String, String> setSessionProperties = stateMachine.getSetSessionProperties();
+        assertThat(setSessionProperties).hasSize(1);
+        assertEquals(setSessionProperties.get(TIME_ZONE_ID), "America/Los_Angeles");
+    }
+
+    @Test
+    public void testSetTimeZoneStringLiteralInvalidZoneId()
+    {
+        QueryStateMachine stateMachine = createQueryStateMachine("SET TIME ZONE 'Matrix/Zion'");
+        SetTimeZone setTimeZone = new SetTimeZone(Optional.of(new StringLiteral("Matrix/Zion")));
+        assertThatThrownBy(() -> executeSetTimeZone(setTimeZone, stateMachine))
+                .isInstanceOf(TimeZoneNotSupportedException.class);
+    }
+
+    @Test
+    public void testSetTimeZoneIntervalLiteral()
+    {
+        QueryStateMachine stateMachine = createQueryStateMachine("SET TIME ZONE INTERVAL '10' HOUR");
+        SetTimeZone setTimeZone = new SetTimeZone(Optional.of(new IntervalLiteral("10", POSITIVE, HOUR)));
+        executeSetTimeZone(setTimeZone, stateMachine);
+
+        Map<String, String> setSessionProperties = stateMachine.getSetSessionProperties();
+        assertThat(setSessionProperties).hasSize(1);
+        assertEquals(setSessionProperties.get(TIME_ZONE_ID), "+10:00");
+    }
+
+    @Test
+    public void testSetTimeZoneIntervalLiteralInvalidZoneId()
+    {
+        QueryStateMachine stateMachine = createQueryStateMachine("SET TIME ZONE INTERVAL '100' HOUR");
+        SetTimeZone setTimeZone = new SetTimeZone(Optional.of(new IntervalLiteral("100", POSITIVE, HOUR)));
+        assertThatThrownBy(() -> executeSetTimeZone(setTimeZone, stateMachine))
+                .isInstanceOf(TrinoException.class)
+                .hasMessage("Invalid offset minutes 6000");
+    }
+
+    @Test
+    public void testSetTimeIntervalLiteralZoneHourToMinute()
```

Comment:
With the additional validation added to the `SetTimeZone` constructor this scenario can't be tested anymore in `TestSetTimeZoneTask`.

I'm adding the mentioned tests to the method `io.trino.sql.parser.TestSqlParser#testSetTimeZone`

Reply:
> Also, it would be good to have tests documenting the boundary supported values (+- 14h).

I added the tests:

- testSetTimeZoneIntervalLiteralGreaterThanOffsetTimeZoneMax
- testSetTimeZoneIntervalLiteralGreaterThanOffsetTimeZoneMin

in the class `io.trino.execution.TestSetTimeZoneTask`